src/Sulu/Bundle/AdminBundle/Resources/js/views/FormOverlayList/FormOverlayList.js

```diff
@@ -0,0 +1,199 @@
+    buildFormStoreOptions(
+        apiOptions: Object,
+        attributes: Object,
+        routerAttributesToFormStore: {[string | number]: string}
+    ) {
+        const formStoreOptions = apiOptions ? apiOptions : {};
+
+        routerAttributesToFormStore = toJS(routerAttributesToFormStore);
+        Object.keys(routerAttributesToFormStore).forEach((key) => {
+            const attributeName = routerAttributesToFormStore[key];
+            const formOptionKey = isNaN(key) ? key : routerAttributesToFormStore[key];
+
+            formStoreOptions[formOptionKey] = attributes[attributeName];
+        });
+
+        return formStoreOptions;
+    }
+
+    setFormRef = (formRef: ?Form) => {
+        this.formRef = formRef;
+    };
+
+    setListRef = (listRef: ?List) => {
+        this.listRef = listRef;
+    };
+
+    componentWillUnmount() {
+        this.destroyFormOverlay();
+    }
+
+    renderFormOverlay() {
+        const {
+            router: {
+                route: {
+                    options: {
+                        addOverlayTitle,
+                        editOverlayTitle,
+                    },
+                },
+            },
+        } = this.props;
+
+        if (!this.formStore) {
+            return null;
+        }
+
+        const overlayTitle = this.formStore.id
+            ? translate(editOverlayTitle || 'sulu_admin.edit')
+            : translate(addOverlayTitle || 'sulu_admin.create');
+
+        return (
+            <Overlay
+                confirmDisabled={!this.formStore.dirty}
+                confirmLoading={this.formStore.saving}
+                confirmText={translate('sulu_admin.save')}
+                onClose={this.handleFormOverlayClose}
+                onConfirm={this.handleFormOverlayConfirm}
+                open={!!this.formStore}
+                size="small"
+                title={overlayTitle}
+            >
+                <div className={formOverlayListStyles.form}>
+                    <ErrorSnackbar
+                        onCloseClick={this.handleErrorSnackbarClose}
+                        visible={!!this.formErrors.length}
+                    />
+                    <Form
+                        onSubmit={this.handleFormSubmit}
+                        ref={this.setFormRef}
+                        store={this.formStore}
+                    />
+                </div>
+            </Overlay>
+        );
+    }
+
+    render() {
+        const {
+            router: {
+                route: {
+                    options: {
+                        formKey,
+                    },
+                },
+            },
+        } = this.props;
+
+        return (
+            <Fragment>
+                <List
+                    {...this.props}
+                    onItemAdd={formKey && this.handleItemAdd}
+                    onItemClick={formKey && this.handleItemClick}
+                    ref={this.setListRef}
+                />
+                {this.renderFormOverlay()}
```

Comment:
I don't really like these additional `render` methods :see_no_evil: I think the code should be easy enough to be just inlined here (which I would say is the case here) or if the code gets more complex I would build a separate react component anyway.